internal/services/network/virtual_network_gateway_resource.go

```diff
@@ -786,6 +777,107 @@ func resourceVirtualNetworkGatewayRead(d *pluginsdk.ResourceData, meta interface
 	return nil
 }
 
+func resourceVirtualNetworkGatewayUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.VirtualNetworkGateways
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	log.Printf("[INFO] preparing arguments for AzureRM Virtual Network Gateway update.")
+
+	id, err := virtualnetworkgateways.ParseVirtualNetworkGatewayID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	existing, err := client.Get(ctx, *id)
+	if err != nil {
+		return err
+	}
+
+	if existing.Model == nil {
+		return fmt.Errorf("retrieving %s: `model` was nil", id)
+	}
+
+	payload := existing.Model
+
+	if d.HasChange("enable_bgp") {
+		payload.Properties.EnableBgp = pointer.To(d.Get("enable_bgp").(bool))
+	}
```

Comment:
Since we're setting information into properties should we also check that it isn't nil 
```suggestion
	if existing.Model == nil {
		return fmt.Errorf("retrieving %s: `model` was nil", id)
	}
	if existing.Model.Properties == nil {
		return fmt.Errorf("retrieving %s: `properties` was nil", id)
	}

	payload := existing.Model

	if d.HasChange("enable_bgp") {
		payload.Properties.EnableBgp = pointer.To(d.Get("enable_bgp").(bool))
	}
```

Reply:
I believe we're ok without this as I don't think Properties can't be nil

<img width="634" alt="image" src="https://github.com/hashicorp/terraform-provider-azurerm/assets/6515518/b7e36753-66ab-4479-95a4-3580eef8afa8">